### PR TITLE
v3.0.3: OSP Fixes and Cleanups

### PR DESCRIPTION
**Pull request by anupsv**
**Changes:**
* One step proof cleanups (i.e, length/kzg verification fixes)
* Updated testing
* Refactor existing tests to be compatible with CI

### CODE REVIEW

File: src/osp/OneStepProverHostIo.sol [hunk 1]
```diff
@@ -31,6 +32,56 @@ contract OneStepProverHostIo is IOneStepProver {
...
+    // This is the point at index 1, since index 0 is the generator value of the G1 group.
+    BN254.G1Point private ALPHA_G1 = BN254.G1Point(5421624913032980671919055010798735843841011930764711817607050648427876929258, 12995821280260994872112541311010834261076556242291585164372488699033268245381);
+
+    function computeGamma(uint256 z, uint256 y, uint256[2] memory p, uint256[4] memory alpha_minus_z_g2) internal pure returns (uint256) {
+        // Encode the variables and compute the keccak256 hash
+        bytes32 hash = keccak256(abi.encodePacked(z, y, p[0], p[1], alpha_minus_z_g2[0], alpha_minus_z_g2[1], alpha_minus_z_g2[2], alpha_minus_z_g2[3]));
+        return uint256(hash) % BN254.FR_MODULUS;
+    }
+
+    //  e((P - y) + gamma . (alpha - z), G2) = e((Q + gamma), (alpha - z))
```
Comment (bxue-l2) [inline src/osp/OneStepProverHostIo.sol]: probably note, the last alpha - z is a passed in

File: src/osp/OneStepProverHostIo.sol [hunk 2]
```diff
@@ -132,7 +183,7 @@ contract OneStepProverHostIo is IOneStepProver {
         19103219067921713944291392827692070036145651957329286315305642004821462161904;
 
     // see: https://github.com/Layr-Labs/eigenda/blob/master/disperser/apiserver/server.go#L35
-    uint256 internal constant eigenDAMaxFieldElementsPerBlob = 268435456;
+    uint256 internal constant eigenDAMaxFieldElementsPerBlob = (2 * 1024 * 1024) / 32;
```
Comment (bxue-l2) [inline src/osp/OneStepProverHostIo.sol]: why we need this variable?

File: src/osp/OneStepProverHostIo.sol [hunk 3]
```diff
@@ -254,49 +305,45 @@ contract OneStepProverHostIo is IOneStepProver {
...
-                });
+                require(proofUint256[0] < BN254.FP_MODULUS, "PROOF_X_LARGER_THAN_FIELD");
+                require(proofUint256[1] < BN254.FP_MODULUS, "PROOF_Y_LARGER_THAN_FIELD");
 
-                BN254.G1Point memory kzgCommitment =
-                    BN254.G1Point(uint256(bytes32(kzgProof[224:256])), uint256(bytes32(kzgProof[256:288])));
+                require(z < BN254.FR_MODULUS, "Z_LARGER_THAN_FIELD");
+                require(y < BN254.FR_MODULUS, "Y_LARGER_THAN_FIELD");
+
+                require((keccak256(kzgProof[224:288]) == bytes32(kzgProof[:32])) && (bytes32(kzgProof[:32]) == leafContents), "KZG_PROOF_WRONG_HASH");
```
Comment (bxue-l2) [inline src/osp/OneStepProverHostIo.sol]: I remembered we hashed datalength into the data too. why is it correct here

File: src/osp/OneStepProverHostIo.sol [hunk 4]
```diff
@@ -254,49 +305,45 @@ contract OneStepProverHostIo is IOneStepProver {
...
-
             // If preimageOffset is greater than or equal to the blob size, leave extracted empty and call it here.
             if (preimageOffset < preimageLength) {
                 // preimageOffset was required to be 32 byte aligned above
                 uint256 tmp = preimageOffset / 32;
-
-                // First, we get the root of unity of order 2**fieldElementsPerBlob.
+                                // First, we get the root of unity of order 2**fieldElementsPerBlob.
                 // We start with a root of unity of order 2**32 and then raise it to
                 // the power of (2**32)/fieldElementsPerBlob to get root of unity we need.
```
Comment (bxue-l2) [inline src/osp/OneStepProverHostIo.sol:393]: maybe just remove comments, as they are not for us

File: src/osp/OneStepProverHostIo.sol (hunk 3 above)
Comment (anupsv) [reply src/osp/OneStepProverHostIo.sol]: It appears that the place where we hash data and length isn't referenced here and seems to be isolated away. In this code it appears that the data for the msp tree is from the other way the hash is created. This is the same way for 4844 from what I can tell. Interesting part is, even if we match the hash data to each other, there is still no connection between the 2 systems on the contract level. Maybe something offchain does.

File: src/osp/OneStepProverHostIo.sol (hunk 1 above)
Comment (anupsv) [reply src/osp/OneStepProverHostIo.sol]: added.

File: src/osp/OneStepProverHostIo.sol (hunk 2 above)
Comment (anupsv) [reply src/osp/OneStepProverHostIo.sol]: The change wasn't specific to the new changes mode. Looks residual and it's no longer used. Removing.

**Pull request merged**